Detectors/TPC/reconstruction/include/TPCReconstruction/KrBoxClusterFinder.h

```diff
@@ -0,0 +1,166 @@
+///    o i i i o
+///    o o o o o
+///
+/// The outer pad-time cells are only addded if the inner cell has signal. For
+/// horizonal vertically aligned inner cells we test like this:
+///        o
+///        i
+///    o i C i o
+///        i
+///        o
+/// For the diagonal cells we check like this.
+///    o o   o o
+///    o i   i o
+///        C
+///    o i   i o
+///    o o   o o
+///
+/// The requirements for a local maxima is:
+/// Charge in bin is >= mQThresholdMax ADC channels.
+/// Charge in bin is larger than all digits in the box around it.
+/// (in the case when it is similar only one of the clusters will be made, see
+/// in the code)
+/// At least mMinNumberOfNeighbours neighbours (= direct neighbours) has a signal.
+///
+/// Implementation:
+/// The RAW data is "expanded" for each sector and stored in a big signal
+/// mMapOfAllDigits = 3d vector of all digits
+///
+/// To make sure that one never has to check if one is inside the sector or not
+/// the arrays are larger than a sector. For the size in row-direction, a constant is used.
+/// For time and pad direction, a number is set (here is room for improvements).
+///
+/// When data from a new sector is encountered, the method
+///
+/// How to use:
+/// Load tpcdigits.root
+/// Loop over all events
+/// Loop over all sectors
+/// Use KrBoxClusterFinder(sector) to create 3D map
+/// Use localMaxima() to find..well the local maxima
+/// localMaxima() yields a vector of a tuple with three int
+/// Loop over this vector and give the tuple to buildCluster()
+/// Write to tree or something similar
+
+#ifndef ALICEO2_TPC_KrBoxClusterFinder_H_
+#define ALICEO2_TPC_KrBoxClusterFinder_H_
+
+//#pragma link C++ class std::vector < KrCluster> + ;
+
+#include "TCanvas.h"
+#include "TColor.h"
+#include "TFile.h"
+#include "TH1.h"
+#include "TTree.h"
+
+#include "TPCBase/Digit.h"
+#include "TPCReconstruction/KrCluster.h"
+
+#include <array>
+#include <iostream>
+#include <tuple>
+#include <vector>
+
+namespace o2
+{
+namespace tpc
+{
+//
+// struct cluster {
+// public:
+//   float totCharge = 0;
+//   float maxCharge = 0;
+//   int size = 0;
+//   // means and sigmas are weighted averages/sigmas
+//   float meanPad = 0;
+//   float meanRow = 0;
+//   float meanTime = 0;
+//   float sigmaPad = 0;
+//   float sigmaRow = 0;
+//   float sigmaTime = 0;
+//   int sector = 0;
+// };
+
+/// KrBoxClusterFinder class can be used to analyze X-ray and Krypton data
+///
+/// It creates a 3D map of all digits in one event. Afterwards the Function
+/// findLocalMaxima() can be used to get the coordinates of all local maxima.
+/// Around these maxima, the cluster is then built with the function
+/// buildCluster. In a box around the center (defined by maxClusterSize....) it
+/// looks for digits that can be assigned to the cluster.
+class KrBoxClusterFinder
+{
+ public:
+  /// Constructor:
+  explicit KrBoxClusterFinder(std::vector<o2::tpc::Digit>* eventSector); // Creates 3D Map

```

Comment:
Do we need a pointer here?
Do we need to fill the digit structures in the constructor? How about a function
```c++
void init(std::vector<o2::tpc::Digit>& eventSector);
```
and move the code in the cxx to this function?

Reply:
I replaced the pointer by a reference:

```suggestion
  explicit KrBoxClusterFinder(std::vector<o2::tpc::Digit>& eventSector);
```
But in my opinion, a init-function is not the best choice here, since the constructor is used to create the object. If you insist on the init-function, I will of course implement it.